internal/server/proto/server.proto

```diff
@@ -287,6 +287,53 @@ message Application {
   string file_change_signal = 3;
 }
 
+// Variable stores a variable's value, and metadata to allow for precedence 
+// sorting and source-specific error messaging 
+message Variable {
+  string name = 1;
+
+  oneof value {
+    string str = 2;
+    string hcl = 3;
+  }
+
+  // Source tracks where a value came from for more helpful error messaging.
+  // This is optional. If unset, we will give a generic error message.
+  oneof source {
+    SourceCLIArg cliarg = 4;
+    SourceCLIFile file = 5;
+    SourceEnvVar envvar= 6;
+    SourceRepoFile repofile = 7;
+    SourceServer server = 8;
```

Comment:
Just a nitpick on this...

First, I'd remove the `Source` prefix. We don't do that for any of our other `oneof`.

Next, I think we separate multiple words with underscoring? I'm not SURE but I think so. So `cli_arg`, `env_var`, etc.

However, I think some of those don't need to be so long, consider: `cli`, `file`, `env`, `vcs`, `server`.

Reply:
Hmmm yeah I think those are good. The comments are there to be more explicit and that + the names make sense.